templates/_helpers.tpl

```diff
@@ -0,0 +1,218 @@
+the deployment. This should only be called if $.tonicSsoConfig is populated
+*/}}
+{{- define "tonic.sso" -}}
+{{- $provider := (required "tonicSsoConfig.provider is required for SSO" .provider | lower) -}}
+- name: TONIC_SSO_PROVIDER
+  value: {{ quote .provider }}
+{{- if .groupFilter }}
+- name: TONIC_SSO_GROUP_FILTER_REGEX
+  value: {{quote .groupFilter}}
+{{- end }}
+{{- if eq $provider "aws" }}
+{{- include "tonic.sso.aws" . }}
+{{- else if eq $provider "azure" }}
+{{ include "tonic.sso.azure" . }}
+{{- else if eq $provider "duo" }}
+{{ include "tonic.sso.duo" . }}
+{{- else if eq $provider "google" }}
+{{ include "tonic.sso.google" . }}
+{{- else if eq $provider "okta" }}
+{{ include "tonic.sso.okta" . }}
+{{- else if eq $provider "keycloak" }}
+{{ include "tonic.sso.keycloak" . }}
+{{- else if eq $provider "saml" }}
+{{- include "tonic.sso.saml" . }}
+{{- else }}
+{{ fail "Unsupported SSO provider " .provider }}
+{{ end }}
+{{- end -}}
+
+{{- define "tonic.sso.aws" -}}
+{{- if (.metdataXml).url -}}
+- name: TONIC_SSO_SAML_IDP_METADATA_XML_URL
+  value: {{ quote .metadataXml.url }}
+{{- else if (.metdataXml).base64 }}
+- name: TONIC_SSO_SAML_IDP_METADATA_XML_BASE64
+  value: {{ quote .metadataXml.base64 }}
+{{- else if .samlIdpMetadataXml }}
+- name: TONIC_SSO_SAML_IDP_METADATA_XML_BASE64
+  value: {{ quote .samlIdpMetadataXml }}
+{{- else -}}
+{{ fail "Either metadataXml.url, metadataXml.base64 or samlIdpMetadataXml must be provided to configure AWS sso" }}
+{{- end -}}
+{{- end -}}
+
+{{- define "tonic.sso.azure" -}}
+- name: TONIC_SSO_TENANT_ID
+  value: {{ required "tonicSsoConfig.tenantId is required to configure Azure sso" .tenantId | quote }}
+- name: TONIC_SSO_CLIENT_ID
+  value: {{ required "tonicSsoConfig.clientId is required to configure Azure sso" .clientId | quote }}
+{{- if not .clientSecret }}
+{{ fail "tonicSsoConfig.clientSecret is required to configure Azure sso" }}
+{{- end }}
+- name: TONIC_SSO_CLIENT_SECRET
+  valueFrom:
+    secretKeyRef:
+      name: tonic-sso-client-secret
+      key: secret
+      optional: false
+{{- end -}}
+
+{{- define "tonic.sso.duo" -}}
+- name: TONIC_SSO_DOMAIN
+  value: {{ required "tonicSsoConfig.domain is required to configure Duo sso" .domain | quote }}
+- name: TONIC_SSO_CLIENT_ID
+  value: {{ required "tonicSsoConfig.clientId is required to configure Duo sso" .clientId | quote }}
+{{- if not .clientSecret }}
+{{ fail "tonicSsoConfig.clientSecret is required to configure Duo sso" }}
+{{- end }}
+- name: TONIC_SSO_CLIENT_SECRET
+  valueFrom:
+    secretKeyRef:
+      name: tonic-sso-client-secret
+      key: secret
+      optional: false
+{{- end -}}
+
+{{- define "tonic.sso.google" -}}
+- name: TONIC_SSO_CLIENT_ID
+  value: {{ required "tonicSsoConfig.clientId is required to configure Google sso" .clientId | quote }}
+{{- if not .clientSecret }}
+{{ fail "tonicSsoConfig.clientSecret is required to configure Google sso" }}
+{{- end }}
```

Comment:
Added these fails, but the documentation suggests that maybe we support creating the secret externally from the chart? Is that something we want to support?

Reply:
Right now it just takes it from the values file. We could support it being created externally but I don't think its important for this PR

Reply:
Was more concerned that we encouraged folks to do that and if so this would break their stuff. We should watch for issue reports for that.